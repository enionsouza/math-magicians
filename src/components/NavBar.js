import { NavLink } from 'react-router-dom';

const NavBar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator/',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quote/',
      text: 'Quote',
    },
  ];

  return (
    <nav className="nav-menu">
      <h1 className="nav-title">Math Magicians</h1>
      <ul className="nav-menu-list">
        {links.map((link) => (
          <li key={link.id}>
            <NavLink
              to={link.path}
              activeClassName="active-link"
              exact
            >
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default NavBar;
