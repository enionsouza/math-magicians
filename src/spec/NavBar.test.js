import { BrowserRouter as Router } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import NavBar from '../components/NavBar';

describe('NavBar component', () => {
  it('NavBar snapshot test', () => {
    const { asFragment: navBar } = render(
      <Router>
        <NavBar />
      </Router>,
    );

    expect(navBar()).toMatchSnapshot();
  });

  describe('NavBar interaction tests', () => {
    it('adds \'active-link\' class to the selected NavLink', () => {
      render(
        <Router>
          <NavBar />
        </Router>,
      );
      const calculatorLink = screen.getByText('Calculator');
      expect(calculatorLink.className).toEqual('');
      fireEvent.click(calculatorLink);
      expect(calculatorLink.className).toEqual('active-link');
    });
  });
});
