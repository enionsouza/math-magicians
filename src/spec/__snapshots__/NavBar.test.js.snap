// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component NavBar snapshot test 1`] = `
<DocumentFragment>
  <nav
    class="nav-menu"
  >
    <h1
      class="nav-title"
    >
      Math Magicians
    </h1>
    <ul
      class="nav-menu-list"
    >
      <li>
        <a
          aria-current="page"
          class="active-link"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator/"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote/"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
